### DISCUSSION TITLE
Connect on boot problem in 0.7.42 - Fire Stick

### DISCUSSION THREAD

**Discussion by rewob29** [schwabe/ics-openvpn#1565] [Configuration problems]
I am running OpenVPN on an Amazon Fire Stick (details below). It runs really well and was really quite straightforward to set up. However having updated from 0.7.41 to 0.7.42 Connect on boot has stopped working (log below). If I understand correctly this now requires the Always-On VPN Android setting for the app to operate correctly? Unfortunately the Fire Stick is locked down by Amazon so I don’t think this is accessible.

Would it perhaps be possible to have a “legacy” setting which uses the previous Connect on boot method so it can be used on devices like the Fire Stick, whilst keeping the improvement for others? Or is there maybe another solution?

The Fire Stick is used by multiple users in a family most of whom are not technical, so being able to start the tunnel automatically is really important :) - I guess that’s a typical use case for this type of device.

Hope that makes sense? Not sure how many people use OpenVPN on a Fire Stick but I find it invaluable!

Many thanks

Fire Sticks:
I’ve tried this on the following two Fire Sticks with the same results
1.	Fire TV Stick (2nd gen) Fire OS 5.2.9.2 (= Android 5.1)
2	Fire TV Stick 4K, Fire OS 6.2.9.4 (= Android 9) 

Log:
12:17 OpenVPN2.6-icsopenvpn [git.v2.6-master-696-98434109b] armeabi-v7a [SSL (OpenSSL)] [LZO] (LZ4] (EPOLL]  [MH/PKTINFO] [AEAD] built on Nov 3 2022
12:17 library versions: OpenSSL 3.0.7 1 Nov 2022, LZO 2.10
12:17 TCP/UDP: Preserving recently used remote address: [AF_INET] xx.xx.xx.xx:xx
12:17 Could not protect VPN socket
12:17 UDPv4 link local: (not bound)
12:17 UDPV4 link remote: [AF_INET]xx.xx.xx.xx:xx
12:17 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
12:17 IXXXXX] Peer Connection Initiated with [AF_INET]xx.xx.xx.xx:xx
12:17 Error: Android establish) method returned null (Really broken network configuration?)
12:17 Failed to open the tun interface
12:17 Exiting due to fatal error
12:17 ERROR: Cannot open TUN
12:17 MGMT: Got unrecognized command>FATAL:ERROR: Cannot open TUN
12:17 Process exited with exit value 1

**Comment by schwabe** [comment]
That is strange. Did you get a prompt to allow VPN  Connection from OpenVPN on every boot before the change?

**Comment by rewob29** [comment]
Hi Arne, thanks for getting back to me – no there was no prompt on 0.7.41, the connection would just start automatically (see log from 0.7.41 below). Happy to provide any other info/help you need (I used to work in tech back in the day – just a retired nerd now providing tech support to the family 😊!!)


Connected: SUCCESS, 10.8.0.4,86.151.228.111.51597

13:02 OpenVPN 2.6-icsopenvpn [giticsopenvpn/V0.7.40-0-g28cb5982] armeabi-v7a [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on

13:02 library versions: OpenSSL 3.0.5 5 Jul 2022, LZO 2.10

13:02 TCP/UDP: Preserving recently used remote address: AF_INET|86.151.228.111:51597

13:02 UDPv4 link local: (not bound)

13:02 UDPV4 link remote: [AF_INET]86.151.228.111:51597

13:02 (RT-AC87U] Peer Connection Initiated with [AF_INET|86.151.228.111:51597

13:02 WARNING: this configuration may cache passwords in memory - - use the auth-nocache option to prevent this

13:02 Initialization Sequence Completed


From: Arne Schwabe ***@***.***>
Date: Saturday, 10 December 2022 at 12:55
To: schwabe/ics-openvpn ***@***.***>
Cc: rewob29 ***@***.***>, Author ***@***.***>
Subject: Re: [schwabe/ics-openvpn] Connect on boot problem in 0.7.42 - Fire Stick (Discussion #1565)

That is strange. Did you get a prompt to allow VPN Connection from OpenVPN on every boot before the change?

**Comment by rewob29** [comment]
Hi there - I have tried this again with the latest version 0.7.46. I noticed that the option has changed from Connect on Boot to Keep VPN connected but I still cannot get this to work on an Amazon Fire Stick. On enabling I get the message to use the Android Always-ON feature, but clicking OPEN VPN SETTINGS results in the message Cannot launch VPN settings directly, please manually open the settings instead. Unfortunately Amazon lock this down on the Fire Stick :(. Any ideas? Many thanks!

**Comment by rewob29** [comment]
I've reverted to 0.7.41 again to keep connect on boot working. Interestingly I still get the message about the Android Always-On feature when selecting connect on boot, but with this version is works fine!
